scala/private/rule_impls.bzl

```diff
@@ -581,7 +582,7 @@ def _write_executable(ctx, rjars, main_class, jvm_flags, wrapper):
         substitutions = {
             "%classpath%": classpath,
             "%java_start_class%": main_class,
-            "%javabin%": "JAVABIN=%s/%s" % (
+            "%javabin%": "JAVABIN=${JAVABIN:-%s/%s}" % (
```

Comment:
Aren’t we double checking the default? Here we set DEFAULT_JAVA_BIN to either JAVABIN or the old default, then above we set JAVA_BIN to either JAVA_BIN or the default.

Seems like this line shouldn’t need to change.

Also: we seem to have both JAVABIN here and JAVA_BIN above. Should we only have JAVABIN?

Reply:
Should just be JAVABIN your right. I believe these are somewhat independent call sites for the same issue. this is the scala_binary path, the other is some simple wrapper script we use in places. Both of them seemed to hard code the path to the JDK to be the runfiles location. 

I'm not sure of the history of the java wrapper, but it looks like originally from the comment it was about setting up a java path. (Though the java launcher preamble will have that too).  it injects a preamble to control the tty for the repl, and hard codes args for scala_test by the looks. Doesn't do anything for scala_binary i can see?

So both would need to be updated if we can launch java with and without the wrapper i believe.

Reply:
Ok so i wasn't following the 2 levels here properly. your right I was setting it twice... which was just plain broken.

So what is happening, now in my understanding(so correct me if i'm wrong). We use the java_binary wrapper, but we always call the inner script. That means however that `JAVABIN` exports to us from the outside world are actually a little tough to obey nicely, since effectively we always want to re-write `JAVABIN` onto  the wrapper. What i've done now is added another new variable and exported it as part of the `java_binary` injection. Then we pick up that variable to control the java path inside the wrapper script. With this I have (finally) proper success with the remote tests! 

(as always, thanks for the good review, led down the path to making it all actually work nicely)